rusk-recovery/src/state.rs

```diff
@@ -241,14 +241,13 @@ fn generate_empty_state<P: AsRef<Path>>(
         .call::<_, ()>(TRANSFER_CONTRACT, "update_root", &(), u64::MAX)
         .expect("root to be updated after pushing genesis note");
 
-    session
-        .call::<_, Option<StakeData>>(
-            STAKE_CONTRACT,
-            "get_stake",
-            &*DUSK_BLS_KEY,
-            u64::MAX,
-        )
-        .expect("Querying a stake should succeed");
+    let owners = session
+        .call::<_, Vec<BlsPublicKey>>(STAKE_CONTRACT, "owners", &(), u64::MAX)
+        .expect("Querying the stake owners should succeed");
+    assert!(
+        owners.data.is_empty(),
+        "Genesis stake should have no owners"
+    );
```

Comment:
Curiosity. Why assert this?

Reply:
Those are not the owners of the piecrust contract. Those are the bls key that can allow additional stakers (until we remove the allowlist).

The wasm contract has no owners, they are added later. Hence that specific assertion.

Why are we asserting something for each contract? Just to be sure that you can properly query fresh deployed contracts